packages/trace-viewer/src/sw/traceModelBackends.ts

```diff
@@ -28,11 +28,11 @@ export class ZipTraceModelBackend implements TraceModelBackend {
   private _entriesPromise: Promise<Map<string, zip.Entry>>;
   private _traceURL: string;
 
-  constructor(traceURL: string, progress: Progress) {
+  constructor(traceURL: string, server: TraceViewerServerBackend, progress: Progress) {
     this._traceURL = traceURL;
     zipjs.configure({ baseURL: self.location.href } as any);
     this._zipReader = new zipjs.ZipReader(
-        new zipjs.HttpReader(formatUrl(traceURL), { mode: 'cors', preventHeadRequest: true } as any),
+        new zipjs.HttpReader(formatUrl(traceURL, server), { mode: 'cors', preventHeadRequest: true } as any),
```

Comment:
we know that our server supports range requests, so i'd like to try replacing this with `zipjs.HttpRangeReader` in a follow-up. that should save some memory.

Reply:
- When trace is on the third-party server, we cannot be sure about range requests support.
- Does that mean we'll fetch parts of the zip lazily, only when requested? That sounds both good and bad 😄

Reply:
> When trace is on the third-party server, we cannot be sure about range requests support.

Absolutely, but it should already help if we just do it for our own trace viewer server.

> Does that mean we'll fetch parts of the zip lazily, only when requested? That sounds both good and bad 😄

That's what I hope it means, yes. My hope is that it fetches only the directory upfront, and then uses the range numbers from there to fetch all files lazily. Right now, the `HttpReader` needs to keep the entire zip in memory, even the parts it never reads.